File: RUBY/sqrt.rb
def sqrt(n)
  return n if n == 0 || n == 1
  $start = 1
  $last = n
  $ans
  while $start <= $last do
    $mid = ($start + $last)/2
    # To check if n is a perfect square
    return $mid if $mid * $mid == n

    # Since I need a floor, I update
    # answer when $mid*$mid is smaller
    # than n, and move closer to sqrt(n)
    if $mid * $mid < n 
      $start = $mid + 1
      $ans = $mid
      # if $mid*$mid is 
      # greater than n
    else
      $last = $mid - 1
    end
  end
  return $ans
end

puts sqrt(25)

puts sqrt(7056)

File: PHP/sqrt.php
<?php
// A PHP program to find floor(sqrt(x)

// Returns floor of
// square root of x		
function floorSqrt($x)
{
	// Base cases
	if ($x == 0 || $x == 1)
	return $x;

	// Do Binary Search
	// for floor(sqrt(x))
	$start = 1; $end = $x; $ans;
	while ($start <= $end)
	{
		$mid = ($start + $end) / 2;

		// If x is a perfect square
		if ($mid * $mid == $x)
			return $mid;

		// Since we need floor, we update
		// answer when mid*mid is smaller
		// than x, and move closer to sqrt(x)
		if ($mid * $mid < $x)
		{
			$start = $mid + 1;
			$ans = $mid;
		}
		
		// If mid*mid is
		// greater than x
		else
			$end = $mid-1;	
	}
	return $ans;
}

// Driver Code
$x = 11;
echo floorSqrt($x), "\n";

// This code is contributed by ajit
?>
